primitives/transaction-pool/src/pool.rs

```diff
@@ -100,6 +100,8 @@ pub enum TransactionStatus<Hash, BlockHash> {
 	/// Transaction has been included in block with given hash.
 	#[serde(rename = "finalized")] // See #4438
 	InBlock(BlockHash),
+	/// Transaction has been finalized by a finality-gadget, e.g GRANDPA
```

Comment:
I think it doesn't have to be necessarily finality gadget (as the consensus might have finality builtin and not rely on a gadget), but the comment is fine, as the majority of the cases it's going to be a gadget.